# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium-shim/patches/with_update/revert_remove_BrokerDuplicateHandle.patch
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium-shim/patches/with_update/revert_remove_BrokerDuplicateHandle.patch
--- a/security/sandbox/chromium-shim/patches/with_update/revert_remove_BrokerDuplicateHandle.patch	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium-shim/patches/with_update/revert_remove_BrokerDuplicateHandle.patch	Wed Jun 12 11:10:48 2019 +0100
@@ -514,7 +514,7 @@
 diff --git a/security/sandbox/chromium/sandbox/win/src/sandbox.h b/security/sandbox/chromium/sandbox/win/src/sandbox.h
 --- a/security/sandbox/chromium/sandbox/win/src/sandbox.h
 +++ b/security/sandbox/chromium/sandbox/win/src/sandbox.h
-@@ -136,14 +136,28 @@ class TargetServices {
+@@ -136,16 +136,30 @@ class TargetServices {
    // processing any untrusted data or running third-party code. If this call
    // fails the current process could be terminated immediately.
    virtual void LowerToken() = 0;
@@ -537,26 +537,28 @@
 +                                     HANDLE* target_handle,
 +                                     DWORD desired_access,
 +                                     DWORD options) = 0;
+ 
+  protected:
+   ~TargetServices() {}
  };
  
  }  // namespace sandbox
  
  
- #endif  // SANDBOX_WIN_SRC_SANDBOX_H_
 diff --git a/security/sandbox/chromium/sandbox/win/src/sandbox_policy.h b/security/sandbox/chromium/sandbox/win/src/sandbox_policy.h
 --- a/security/sandbox/chromium/sandbox/win/src/sandbox_policy.h
 +++ b/security/sandbox/chromium/sandbox/win/src/sandbox_policy.h
-@@ -21,27 +21,31 @@ class TargetPolicy {
+@@ -24,27 +24,31 @@ class TargetPolicy {
    // exactly like the CreateProcess API does. See the comment at the top of
    // process_thread_dispatcher.cc for more details.
    enum SubSystem {
-     SUBSYS_FILES,             // Creation and opening of files and pipes.
-     SUBSYS_NAMED_PIPES,       // Creation of named pipes.
-     SUBSYS_PROCESS,           // Creation of child processes.
-     SUBSYS_REGISTRY,          // Creation and opening of registry keys.
-     SUBSYS_SYNC,              // Creation of named sync objects.
-+    SUBSYS_HANDLES,           // Duplication of handles to other processes.
-     SUBSYS_WIN32K_LOCKDOWN    // Win32K Lockdown related policy.
+     SUBSYS_FILES,           // Creation and opening of files and pipes.
+     SUBSYS_NAMED_PIPES,     // Creation of named pipes.
+     SUBSYS_PROCESS,         // Creation of child processes.
+     SUBSYS_REGISTRY,        // Creation and opening of registry keys.
+     SUBSYS_SYNC,            // Creation of named sync objects.
++    SUBSYS_HANDLES,         // Duplication of handles to other processes.
+     SUBSYS_WIN32K_LOCKDOWN  // Win32K Lockdown related policy.
    };
  
    // Allowable semantics when a rule is matched.